### PR TITLE
processConstraints modified to increase performance

### PR DESCRIPTION
**Pull request by xavi-pinsach**

### CODE REVIEW

**Review by phated — COMMENTED** [review]
Looks fine to me, but we'll need Jordi to look because I don't understand the math stuff

File: package.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
     "ffjavascript": "0.2.55",
     "js-sha3": "^0.8.0",
     "logplease": "^1.2.15",
-    "r1csfile": "0.0.36"
+    "r1csfile": "0.0.40",
+    "readline": "^1.3.0"
```
Comment (phated) [inline package.json:51]:
```suggestion
```
I've removed this dependency because it is not correct to include it.

**Review by phated — CHANGES_REQUESTED** [review]
An incorrect dependency was added in the merge

File: package.json (hunk 1 above)
Comment (xavi-pinsach) [reply package.json]: done!

Comment (phated) [reply package.json]: Thanks for cleaning that up! ❤️

**Pull request merged**